docs/Reporting.md

```diff
@@ -20,6 +21,25 @@ The report will look something like:
 
 ![Cypress Image Diff Report](../report-example.png)
 
+### Generate Custom HTML report
+If you want to generate your custom report, pass a builder function to `HTML_REPORTER` in [custom config file](https://github.com/kien-ht/cypress-image-diff#custom-config-file). In the below example, a file named `example.json` will be generated in the current directory after all the tests are run:
+```
+const fs = require('fs-extra')
+
+const config = {
+  ROOT_DIR: '',
+  HTML_REPORTER: (testResults) => {
+    fs.writeFile('./example.json', JSON.stringify(testResults, null, 2), (err) => {
```

Comment:
what you think about encapsulating this implementation within the plugin, and just providing an object with the necessary values for it to function?

let me know what you think

Reply:
I think that would be great. So provide configuration options for JSON reports and a hook (after:run hook), right? I think I'll rename the hook from `HTML_REPORTER` to `AFTER_RUN` to be precise. This is a nice hook for users that they can tap into at the end of the process with the `testResults` argument, so they could trigger their custom builder function without reading the test results from the JSON reports. What do you think?

Reply:
I think you're right. Just provide configuration options for JSON reports instead of giving away too many options for users that they could implement themselves. I'll update it shortly.